caffe2/operators/utility_ops.h

```diff
@@ -738,6 +738,105 @@ class ScatterAssignOp : public Operator<Context> {
   INPUT_TAGS(DATA, INDICES, SLICES);
 };
 
+template <class Context>
+class ScatterOp : public Operator<Context> {
+ public:
+  USE_OPERATOR_CONTEXT_FUNCTIONS;
+
+  template <class... Args>
+  explicit ScatterOp(Args&&... args)
+      : Operator<Context>(std::forward<Args>(args)...),
+        OP_SINGLE_ARG(int, "axis", axis_, 1) {
+  }
+
+  virtual ~ScatterOp() noexcept {}
+
+  bool RunOnDevice() override {
+    return DispatchHelper<TensorTypes<int32_t, int64_t>>::call(
+        this, this->template Input<Tensor>(INDICES, CPU));
```

Comment:
Should we fix CPU here?

Reply:
I would suggest to provide better information here, if the tensor is on some devices other than CPU. To explicitly tell them, Scatter only supports CPU at the moment. :-)

Reply:
If it is a CPU only op, why not remove template Context?

Reply:
added a check on the device type, and changed Operator<Context> to Operator<CPUContext>.